src/components/cell-tooltip/CellTooltipText.js

```diff
@@ -0,0 +1,49 @@
+import React, { useLayoutEffect, useRef } from 'react';
+
+export default function CellTooltipText(props) {
+  const {
+    x,
+    y,
+    factors,
+  } = props;
+
+  const el = useRef(null);
+  const offsetPercentage = 10;
+
+  // Do collision detection based on the bounds of the parent (.card-body) element.
+  useLayoutEffect(() => {
+    const rect = el.current.parentNode.getBoundingClientRect();
+    const { width, height } = rect;
+    if (x > width / 2) {
+      el.current.style.transform = `translateX(-${100 + offsetPercentage}%)`;
+    } else {
+      el.current.style.transform = `translateX(${offsetPercentage}%)`;
+    }
+    if (y > height / 2) {
+      el.current.style.transform = `${el.current.style.transform} translateY(-${100 + offsetPercentage}%)`;
+    } else {
+      el.current.style.transform = `${el.current.style.transform} translateY(${offsetPercentage}%)`;
+    }
```

Comment:
- Prefer not to reassign variables, usually.
- Since the target of the assignment is the same, maybe a ternary?:
```javascript
const translateX = (x > width / 2) ? -(100 + offsetPercentage) : offsetPercentage;
const translateY = (x > height / 2) ? -(100 + offsetPercentage) : offsetPercentage;
el.current.style.transform = `translateX(${translateX}%); translateY(${translateY}%);`
```

Reply:
Thanks, this does seem much nicer.